store/tikv/coprocessor.go

```diff
@@ -416,22 +417,33 @@ const minLogCopTaskTime = 300 * time.Millisecond
 func (worker *copIteratorWorker) run(ctx context.Context) {
 	defer worker.wg.Done()
 	for task := range worker.taskCh {
-		respCh := worker.respChan
-		if respCh == nil {
-			respCh = task.respChan
-		}
+		func() {
+			defer func(t *copTask) {
```

Comment:
don't defer here.
defer in the handle function...
The dispatch loop should be as simple as possible.